### ISSUE TITLE
Method defined with definePropery is not callable

### ISSUE THREAD

**Issue by ilogico** [microsoft/TypeScript#43319]
# Bug Report

### 🔎 Search Terms
defineProperty callable

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.2.3&useJavaScript=true#code/GYVwdgxgLglg9mABAQQBQEpEG8BQBfHHAeQCMArAU2gDoATC4GMCgBQCc4AHCtqAT1TJqnDlDj9uAGkQAiALYzpWRADcAhgBsQFAFyJQkWAkRyM2PNIDubGFDUkNuxFDbbpEBIwDmINvcd6LtqIeOgA3DgeYADOUIhqiAC8iMyWKBgROGrUpuiEhEA)

### 💻 Code

```js
// This is JavaScript, not TypeScript
function A() {} // define A with class syntax leads to the same results

Object.defineProperty(A.prototype, "m", { value: function m() {}, writable: true, configurable: true });
const a = new A();

a.m()
```

### 🙁 Actual behavior
TypeScript identifies `m` as a method, but says it has no call signatures: Type '{}' has no call signatures.

### 🙂 Expected behavior
TypeScript allows `a.m()`.

**Comment by NightshadeX9X** [comment]
I don't think you understand. TypeScript doesn't check properties on values (to generate types) at run time. At compile-time, it simply checks any reads/writes to that variable based on the type you assigned it or that was inferred. It cannot understand any variable manipulations you perform at runtime.
